### ISSUE TITLE
Unable to copy/paste a URL to share with someone

### ISSUE THREAD

**Issue by major** [redhatcloudx/cloud-image-directory-frontend#53]
I went through these steps to simulate sharing a URL with someone else:

1. Go to [imagedirectory.cloud]
2. Click **AWS** at the top
3. Copy URL to clipboard: `https://imagedirectory.cloud/browser/AWS`
4. Open a new tab
5. Paste URL into the tab
6. Result:

```xml
<Error>
<Code>NoSuchKey</Code>
<Message>The specified key does not exist.</Message>
<Key>browser/AWS</Key>
<RequestId>TAZTVDRSKJCTNYP9</RequestId>
<HostId>
tVpUYTTkuI3zkCHC8fjRU4cQnXII6MTMglCjoPHF29jqqlPswT5BRRpXQn/l6z88ognoXL7sSpO0/71IXEEGPpzMmi75HarzkfoYoMNfI3s=
</HostId>
</Error>
```

The same thing happens if you click **AWS** and refresh the page.

**Comment by ondrejbudai** [comment]
Hit the same issue with the following reproducer:

1) Go to [imagedirectory.cloud]
2) Click AWS at the top
3) Refresh the page

**Comment by FKolwa** [comment]
@major @ondrejbudai @miyunari 
So this is interesting. The code does what it's supposed to be doing. The problem is that CloudFront doesn't understand single page applications. I've replicated our page on a different bucket and connected the CDN: https://d30uz1n5lx6gxn.cloudfront.net/

To solve the issue I had to create a custom error page response in CloudFront that redirects 404 errors to the index.html file and let the react router handle the error reporting instead. This way the path you are trying to access is interpreted by the router as usual and the correct subpage is returned: https://d30uz1n5lx6gxn.cloudfront.net/browser/GCP 

And now we finally get the appropriate 404 page when a page doesn't exist: https://d30uz1n5lx6gxn.cloudfront.net/browser/doesnotexist

I'm going ahead and create a fix on the infrastructure repo.
